test/release/examples/primers/forallLoops.chpl

```diff
@@ -0,0 +1,305 @@
+"Must-parallel" forall expression
+---------------------------------
+
+The following forall-expression produces new values in parallel.
+We store these values in a new array.
+*/
+
+var B = forall a in A do a * 3;
+
+writeln("After initialization, B is:");
+writeln(B);
+writeln();
+
+/*
+Zippered "must-parallel" forall statement
+-----------------------------------------
+
+We can iterate over multiple arrays in parallel with a zippered loop.
+Here we illustrate zippering arrays and domains:
+*/
+
+var C: [1..n] real;
+forall (a, b, i) in zip(A, B, C.domain) do
+  C[i] = a * 10 + b / 10;
+
+writeln("After a zippered loop, C is:");
+writeln(C);
+writeln();
+
+/*
+"May-parallel" forall statement
+-------------------------------
+
+The iterator ``onlySerial`` defined below does not have any parallel
+forms. Therefore the may-parallel loop ``[i in onlySerial(n)]``
+will accept it, executing its iterations serially:
+*/
+
+iter onlySerial(m: int) {
+  for j in 1..m do
+    yield j;
+}
+
+[i in onlySerial(n)] {
+  writeln("in onlySerial iteration #", i);
+}
+writeln();
+
+/*
+Using the following must-parallel loop instead would cause
+a "parallel iterator is not found" error:
+
+  .. code-block:: chapel
+
+      forall i in onlySerial(n) {
+        writeln("in iteration #", i);
+      }
+
+"May-parallel" forall expression
+--------------------------------
+
+Given that these arrays provide parallel iteration, the following
+may-parallel expression will be computed in parallel:
+*/
+
+var D = [(a,b,c) in zip(A,B,C)] a + c - b;
+
+writeln("The result of may-parallel expression, D is:");
+writeln(D);
+writeln();
+
+/*
+As with must-parallel loops, if A were a distributed array,
+iteration locality while computing the above expression
+would be determined by A's domain map.
+
+Task Intents and Shadow Variables
+---------------------------------
+
+A forall loop may name some variables declared outside the loop,
+or "outer variables". If so, "shadow variables" are introduced.
+Each task created by the parallel iterator gets its own set of
+shadow variables, one per outer variable.
+
+ - Each shadow variable behaves as if it were a formal argument
+   of the task function for the task. The outer variable is passed
+   to this formal argument according to the argument intent
+   associated with the shadow variable, which is called a "task intent".
+
+ - The name of an outer variable within the lexical scope of the loop
+   body refers implicitly to the corresonding shadow variable. If the
+   parallel iterator causes multiple iterations of the loop to be
+   executed by the same task, these iterations refer to the same set
+   of shadow variables.
+
+ - Each shadow variable is deallocated at the end of its task.
+
+The default argument intent is used by default. For numeric types,
+this implies capturing the value of the outer variable by the time
+the task starts executing. Arrays are passed by reference, and so are
+sync, single, and atomic variables.
+*/
```

Comment:
The comments from the other primer could apply here as well.  Alternatively, you could refer to that section of the taskParallel primer to avoid repeating it here (I'm a little torn between it being nice to have each primer stand alone and having to maintain two copies of very similar text).